tests/test_tokens.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the tokens module."""
+
+import dataclasses
+
+from ghga_service_commons.utils.jwt_helpers import (
+    decode_and_validate_token,
+    generate_jwk,
+)
+
+from wps.core.models import WorkOrderToken, WorkType
+from wps.core.tokens import (
+    generate_work_package_access_token,
+    hash_token,
+    sign_work_order_token,
+)
+
+
+def test_generate_work_package_access_token():
+    """Test the generation of a work package access token."""
+    token = generate_work_package_access_token()
+    assert len(token) == 24
+    assert token.isascii()
+    assert token.isalnum()
+    another_token = generate_work_package_access_token()
+    assert len(another_token) == len(token)
+    assert another_token.isascii()
+    assert another_token.isalnum()
+    assert another_token != token
+
+
+def test_hash_token():
+    """Test hashing of tokens."""
+    token = "abc123" * 4
+    hashed_token = hash_token(token)
+    assert len(hashed_token) == 64
+    assert all(char.isdigit() or "a" <= char <= "f" for char in hashed_token)
```

Comment:
we don't care about that right?

Reply:
Just to make sure it is converted to a plain ascii string and not binary. That we use hexdigest is an implementation detail, right.

Reply:
Changed that in 8dcc098.